### ISSUE TITLE
bug: dgs incompatible with spring cloud and spring data elastic

### ISSUE THREAD

**Issue by menevs** [Netflix/dgs-framework#486]
## Expected behavior

The Application should start without error

## Actual behavior

The apllication crashes at start with the following error:
`***************************
APPLICATION FAILED TO START
***************************

Description:

An attempt was made to call a method that does not exist. The attempt was made from the following location:

    org.springframework.boot.env.OriginTrackedYamlLoader.createYaml(OriginTrackedYamlLoader.java:69)

The following method did not exist:

    'void org.yaml.snakeyaml.LoaderOptions.setMaxAliasesForCollections(int)'

The method's class, org.yaml.snakeyaml.LoaderOptions, is available from the following locations:

    jar:file:/home/menevs/.m2/repository/org/yaml/snakeyaml/1.23/snakeyaml-1.23-android.jar!/org/yaml/snakeyaml/LoaderOptions.class
    jar:file:/home/menevs/.m2/repository/org/yaml/snakeyaml/1.27/snakeyaml-1.27.jar!/org/yaml/snakeyaml/LoaderOptions.class

The class hierarchy was loaded from the following locations:

    org.yaml.snakeyaml.LoaderOptions: file:/home/menevs/.m2/repository/org/yaml/snakeyaml/1.23/snakeyaml-1.23-android.jar

Action:

Correct the classpath of your application so that it contains a single, compatible version of org.yaml.snakeyaml.LoaderOptions

ERROR LoggingFailureAnalysisReporter 

***************************
APPLICATION FAILED TO START
***************************

Description:

An attempt was made to call a method that does not exist. The attempt was made from the following location:

    org.springframework.boot.env.OriginTrackedYamlLoader.createYaml(OriginTrackedYamlLoader.java:69)

The following method did not exist:

    'void org.yaml.snakeyaml.LoaderOptions.setMaxAliasesForCollections(int)'

The method's class, org.yaml.snakeyaml.LoaderOptions, is available from the following locations:

    jar:file:/home/menevs/.m2/repository/org/yaml/snakeyaml/1.23/snakeyaml-1.23-android.jar!/org/yaml/snakeyaml/LoaderOptions.class
    jar:file:/home/menevs/.m2/repository/org/yaml/snakeyaml/1.27/snakeyaml-1.27.jar!/org/yaml/snakeyaml/LoaderOptions.class

The class hierarchy was loaded from the following locations:

    org.yaml.snakeyaml.LoaderOptions: file:/home/menevs/.m2/repository/org/yaml/snakeyaml/1.23/snakeyaml-1.23-android.jar

Action:

Correct the classpath of your application so that it contains a single, compatible version of org.yaml.snakeyaml.LoaderOptions

Process finished with exit code 1
`

## Steps to reproduce

Create a new Spring Boot App with the latest Versions of Spring Boot and Spring Data ElasticSearch or Spring Cloud Kubernetes (or any other Framework that uses snakeyaml > 1.23) and also add the dgs framework.

## Notes

For me it seems that graphql-dgs-mocking uses javafaker which uses snakeyaml 1.23 which might be incompatible with newer versions. As it looks like the javafaker dependency is no longer in active development there are no dependency upgrades. I tried to exclude dgs-mocking or javafaker in my pom files but it didn't worked (don't know what maven did to readd them or I made a mistake)
IntelliJ shows the conflict within the dependency analyzer plugin see the attached screenshot:
![image](https://user-images.githubusercontent.com/79131101/125956654-dfd9a8b7-b970-440b-b66d-20248f70c23a.png)

**Comment by menevs** [comment]
Not using the BOM worked for me to exclude the javafaker. But I think it should be fixed in dgs.

My workaround:
```
         <dependency>
            <groupId>com.netflix.graphql.dgs</groupId>
            <artifactId>graphql-dgs-spring-boot-starter</artifactId>
            <version>4.4.0</version>
            <exclusions>
                <exclusion>
                    <artifactId>javafaker</artifactId>
                    <groupId>com.github.javafaker</groupId>
                </exclusion>
            </exclusions>

        </dependency>
```

**Comment by berngp** [comment]
@menevs could you please point me to your POM file?

**Comment by berngp** [comment]
Sent a [PR](https://github.com/DiUS/java-faker/pull/655) to `java-faker` to see if they are willing to shade `snakeyaml`.

**Comment by berngp** [comment]
Another option is to shade `java-faker` and its dependencies in our `graphql-dgs-mocking` module. Now, all these, should only be a problem if you explicitly have a dependency on `graphql-dgs-mocking`.

**Comment by paulbakker** [comment]
I think we should redesign the dependency on the mocking module, so that it becomes optional. This will also pave the path to potentially deprecate the mocking module, since I don't think it gets used a lot.

**Comment by paulbakker** [comment]
This should be fixed by #541
